### PR TITLE
Rename to consistent usage of Ewasm

### LINKED ISSUE #7932: Rename to Ewasm/ewasm
Yul and the source through contains various spellings of Ewasm. The preferred spelling is "Ewasm" and we should use "ewasm" for CLI options.

One example, the machine flag for Yul uses "eWasm".

### PR DESCRIPTION
**Pull request by axic**
Depends on #8054.
Closes #7932 .

### CODE REVIEW

File: libyul/backends/wasm/BinaryTransform.h [hunk 1]
```diff
@@ -15,7 +15,7 @@
 	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
 */
 /**
- * EWasm to binary encoder.
+ * Component that transforms interval Wasm representation to binary.
```
Comment (axic) [inline libyul/backends/wasm/BinaryTransform.h]: interval -> internal

File: libyul/backends/wasm/BinaryTransform.cpp [hunk 2]
```diff
@@ -15,7 +15,7 @@
 	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
 */
 /**
- * EWasm to binary encoder.
+ * Component that transforms interval Wasm representation to binary.
```
Comment (axic) [inline libyul/backends/wasm/BinaryTransform.cpp]: Here too.

File: solc/CommandLineInterface.cpp [hunk 3]
```diff
@@ -125,7 +125,6 @@ static string const g_strYul = "yul";
 static string const g_strYulDialect = "yul-dialect";
 static string const g_strIR = "ir";
 static string const g_strIPFS = "ipfs";
-static string const g_strEWasm = "ewasm";
```
Comment (axic) [inline solc/CommandLineInterface.cpp:128]: The renaming saved one variable!!111! :wink:

**Review by leonardoalt — COMMENTED** [review]
Looks good when the typos that were already mentioned are fixed.

File: libsolidity/interface/CompilerStack.h [hunk 4]
```diff
@@ -338,8 +338,8 @@ class CompilerStack: boost::noncopyable
 		eth::LinkerObject runtimeObject; ///< Runtime object.
 		std::string yulIR; ///< Experimental Yul IR code.
 		std::string yulIROptimized; ///< Optimized experimental Yul IR code.
-		std::string eWasm; ///< Experimental eWasm text representation
-		eth::LinkerObject eWasmObject; ///< Experimental eWasm code
+		std::string Ewasm; ///< Experimental Ewasm text representation
```
Comment (chriseth) [inline libsolidity/interface/CompilerStack.h]:
```suggestion
		std::string ewasm; ///< Experimental Ewasm text representation
```

File: libsolidity/interface/CompilerStack.h [hunk 5]
```diff
@@ -338,8 +338,8 @@ class CompilerStack: boost::noncopyable
...
+		eth::LinkerObject EwasmObject; ///< Experimental Ewasm code
```
Comment (chriseth) [inline libsolidity/interface/CompilerStack.h]:
```suggestion
		eth::LinkerObject ewasmObject; ///< Experimental Ewasm code
```

File: libsolidity/interface/StandardCompiler.cpp [hunk 6]
```diff
@@ -956,11 +956,11 @@ Json::Value StandardCompiler::compileSolidity(StandardCompiler::InputsAndSetting
...
 			contractData["irOptimized"] = compilerStack.yulIROptimized(contractName);
 
-		// eWasm
+		// Ewasm
 		if (compilationSuccess && isArtifactRequested(_inputsAndSettings.outputSelection, file, name, "ewasm.wast", wildcardMatchesExperimental))
-			contractData["ewasm"]["wast"] = compilerStack.eWasm(contractName);
+			contractData["ewasm"]["wast"] = compilerStack.Ewasm(contractName);
 		if (compilationSuccess && isArtifactRequested(_inputsAndSettings.outputSelection, file, name, "ewasm.wasm", wildcardMatchesExperimental))
-			contractData["ewasm"]["wasm"] = compilerStack.eWasmObject(contractName).toHex();
+			contractData["ewasm"]["wasm"] = compilerStack.EwasmObject(contractName).toHex();
```
Comment (chriseth) [inline libsolidity/interface/StandardCompiler.cpp]:
```suggestion
			contractData["ewasm"]["wasm"] = compilerStack.ewasmObject(contractName).toHex();
```

File: solc/CommandLineInterface.cpp [hunk 7]
```diff
@@ -345,23 +344,23 @@ void CommandLineInterface::handleIR(string const& _contractName)
...
 			);
 		}
 		else
 		{
-			sout() << "EWasm text:" << endl;
-			sout() << m_compiler->eWasm(_contractName) << endl;
-			sout() << "EWasm binary (hex): " << m_compiler->eWasmObject(_contractName).toHex() << endl;
+			sout() << "Ewasm text:" << endl;
+			sout() << m_compiler->Ewasm(_contractName) << endl;
+			sout() << "Ewasm binary (hex): " << m_compiler->EwasmObject(_contractName).toHex() << endl;
```
Comment (chriseth) [inline solc/CommandLineInterface.cpp]:
```suggestion
			sout() << "Ewasm binary (hex): " << m_compiler->ewasmObject(_contractName).toHex() << endl;
```

**Comment by chriseth** [comment]
Build fails:
```
/root/project/libsolidity/interface/StandardCompiler.cpp: In member function 'Json::Value dev::solidity::StandardCompiler::compileSolidity(dev::solidity::StandardCompiler::InputsAndSettings)':
/root/project/libsolidity/interface/StandardCompiler.cpp:961:50: error: 'class dev::solidity::CompilerStack' has no member named 'ewasm'; did you mean 'Ewasm'?
    contractData["ewasm"]["wast"] = compilerStack.ewasm(contractName);
                                                  ^~~~~
                                                  Ewasm
/root/project/libsolidity/interface/StandardCompiler.cpp:963:50: error: 'class dev::solidity::CompilerStack' has no member named 'ewasmObject'; did you mean 'EwasmObject'?
    contractData["ewasm"]["wasm"] = compilerStack.ewasmObject(contractName).toHex();
                                                  ^~~~~~~~~~~
                                                  EwasmObject
```

File: libsolidity/interface/CompilerStack.h [hunk 8]
```diff
@@ -251,11 +251,11 @@ class CompilerStack: boost::noncopyable
 	/// @returns the optimized IR representation of a contract.
 	std::string const& yulIROptimized(std::string const& _contractName) const;
 
-	/// @returns the eWasm text representation of a contract.
-	std::string const& eWasm(std::string const& _contractName) const;
+	/// @returns the Ewasm text representation of a contract.
+	std::string const& Ewasm(std::string const& _contractName) const;
```
Comment (chriseth) [inline libsolidity/interface/CompilerStack.h]:
```suggestion
	std::string const& ewasm(std::string const& _contractName) const;
```

File: libsolidity/interface/CompilerStack.h [hunk 9]
```diff
@@ -251,11 +251,11 @@ class CompilerStack: boost::noncopyable
...
 
-	/// @returns the eWasm representation of a contract.
-	eth::LinkerObject const& eWasmObject(std::string const& _contractName) const;
+	/// @returns the Ewasm representation of a contract.
+	eth::LinkerObject const& EwasmObject(std::string const& _contractName) const;
```
Comment (chriseth) [inline libsolidity/interface/CompilerStack.h]:
```suggestion
	eth::LinkerObject const& ewasmObject(std::string const& _contractName) const;
```

**Review by chriseth — APPROVED** [review]

**Pull request merged**